src/control/jobs/Scheduler.h

```diff
@@ -111,25 +114,31 @@ class Scheduler {
 
     GThread* thread = nullptr;
 
-    GCond jobQueueCond{};
-    GMutex jobQueueMutex{};
-
-    GMutex schedulerMutex{};
+    std::condition_variable jobQueueCond{};
+    std::mutex jobQueueMutex{};
+    std::mutex schedulerMutex{};
 
     /**
-     * This is need to be sure there is no job running if we delete a page, else we may access delete memory...
+     * This is need to be sure there is no job running if we delete a page.
+     * If a job is, we may access deleted memory.
      */
-    GMutex jobRunningMutex{};
+    std::mutex jobRunningMutex{};
 
-    GQueue queueUrgent{};
-    GQueue queueHigh{};
-    GQueue queueLow{};
-    GQueue queueNone{};
+    /**
+     * Jobs of each priority. New jobs
+     * are added to the back of each queue.
+     */
+    std::deque<Job*> queueUrgent{}, queueHigh{}, queueLow{}, queueNone{};
```

Comment:
These are `std::deque`s and not `std::queue`s. This is because we use iteration. See https://stackoverflow.com/questions/1259099/stdqueue-iteration.

Reply:
It's good to use std::deque :+1: 
But please don't declare more than one variable of the same type in one line.